conda_forge_tick/update_upstream_versions.py

```diff
@@ -488,45 +42,72 @@ def get_latest_version(
             ver = source.get_version(url)
             logger.debug("ver: %s", ver)
             if ver:
-                return ver
+                version_data["new_version"] = ver
+                return version_data
             else:
-                meta_yaml["bad"] = f"Upstream: Could not find version on {source.name}"
+                logger.debug(f"Upstream: Could not find version on {source.name}")
+                version_data[
+                    "bad"
+                ] = f"Upstream: Could not find version on {source.name}"
         if not meta_yaml.get("bad"):
-            meta_yaml["bad"] = "Upstream: unknown source"
-        return False
+            logger.debug("Upstream: unknown source")
+            version_data["bad"] = "Upstream: unknown source"
+
+        version_data["new_version"] = False
+        return version_data
+
+
+# It's expected that your environment provide this info.
+CONDA_FORGE_TICK_DEBUG = os.environ.get("CONDA_FORGE_TICK_DEBUG", False)
 
 
 def _update_upstream_versions_sequential(
-    gx: nx.DiGraph, sources: Iterable[AbstractSource],
+    gx: nx.DiGraph, sources: Iterable[AbstractSource] = None,
 ) -> None:
+
     _all_nodes = [t for t in gx.nodes.items()]
     random.shuffle(_all_nodes)
 
+    # Inspection the graph object and node update:
+    # print(f"Number of nodes: {len(gx.nodes)}")
+    node_count = 0
     to_update = []
     for node, node_attrs in _all_nodes:
-        attrs = node_attrs["payload"]
-        if attrs.get("bad") or attrs.get("archived"):
-            attrs["new_version"] = False
-            continue
-        to_update.append((node, attrs))
+        with node_attrs["payload"] as attrs:
+            if (attrs.get("bad") and "Upstream" not in attrs["bad"]) or attrs.get(
```

Comment:
```suggestion
            if "Upstream" not in attrs.get("bad", []) or attrs.get(
```

Reply:
Thanks CJ,I did not know that the `.get()` method could be used this way.